### ISSUE TITLE
Improve handling when DJDT views don't respond with JSON

### ISSUE THREAD

**Issue by TreyWW** [jazzband/django-debug-toolbar#1847]
Sidebar loads fine until you press any of the buttons, for example "Cache".

As soon as i press it i get 
![image](https://github.com/jazzband/django-debug-toolbar/assets/73353716/77a6125c-b73f-46f1-ae14-d422187142ea)

```
VM14865:2 Uncaught (in promise) SyntaxError: Unexpected token '<', "
<html>

<h"... is not valid JSON
```

I load django toolbar at the start so

```py
urlpatterns = [

]
if settings.DEBUG:
    import debug_toolbar

    urlpatterns += [
        url("__debug__/", include(debug_toolbar.urls)),
    ]

urlpatterns += [] # my normal urls
```

Also does it after my normal urls. Does it in all of my projects now, not sure why but it's really annoying. One of my other test pages calls a request every 7 seconds, and each time it fetches data i get that error on my screen. So annoying.

I'm happy to provide more details

Thanks :)

---

Edit from Tim:

This issue is because the toolbar views have some authorization logic wrapping them causing the responses to be HTML rather than JSON. Our JS frontend should identify when the responses don't match expectations and notify the user of such to make it easier to diagnose.

**Comment by TreyWW** [comment]
Also error comes from here

![image](https://github.com/jazzband/django-debug-toolbar/assets/73353716/f01f851f-315d-4084-87d9-d6957b74fc7a)

**Comment by TreyWW** [comment]
Sure thing

![image](https://github.com/jazzband/django-debug-toolbar/assets/73353716/a1a840e1-22d2-4ca8-8f9c-110a711b5126)
![image](https://github.com/jazzband/django-debug-toolbar/assets/73353716/9e5e6ab8-ac6c-4b26-8ad2-f8f45f9afb71)



aha, `http://127.0.0.1:8000/login/?next=/__debug__/history_sidebar/%3Fstore_id%3D221f0a9e2ea44d7ba227eddf737e62d7`

By that i assume it's trying to login to view things. I need to find the url name to whitelist that i suppose
I'll let you know if i cant find it

**Comment by tim-schilling** [comment]
As an aside, the preferred terms for that type of functionality are allow-list and deny-list. I know you're not from the US so there are some contextual differences, but whitelist and blacklist have their roots in systematic racism and discrimination in the US.

**Comment by TreyWW** [comment]
Yeah i've added 

```
LOGIN_REQUIRED_IGNORE_PATHS = [
    r"/login/$", r"^__debug__/$", r"debug__/$", r"__debug__/$",
]
```

But still same error

**Comment by tim-schilling** [comment]
Thank you for sharing that. It's because `$` is an indicator the last character in the string. If you want to match anything that starts with just `__debug__/` you need to remove that character.

**Comment by TreyWW** [comment]
` r"__debug__/"` still gives an error too

**Comment by tim-schilling** [comment]
Does it need the leading `/`?

**Comment by tim-schilling** [comment]
Re-opening because I'm re-purposing the issue

**Comment by TreyWW** [comment]
Hmm, i'm attempting to do this fix on my other project and this still doesn't work even with

```py
LOGIN_REQUIRED_IGNORE_PATHS = [
    r"/login/$", r"__debug__"
]
```

**Comment by matthiask** [comment]
Yeah, the middleware uses `re.match`, not `re.search`: 
https://github.com/CleitonDeLima/django-login-required-middleware/blob/d4535f5a9f61d4d24012fbbbd1d32d4d9b7002bf/login_required/middleware.py#L26

Therefore you *have* to use a regex which matches from the beginning, not somewhere inside. 

(You could try `r".*__debug__"` but that might be too broad.)

**Comment by folt** [comment]
I have the same problem.
The example from the documentation results in an error.
`path("__debug__/", include("debug_toolbar.urls"))`

**Comment by tim-schilling** [comment]
@folt I appreciate you voicing your support for this issue. What's would be helpful for this issue is including the actual error to confirm that it is indeed the error that this issue is regarding. If it's not the same error, then this is not the appropriate place for that message.

**Comment by folt** [comment]
On some of my projects I get this error. But it did not block their work and I rarely include django-debug-toolbar in the production environment. That's why I always ignored this error.

After doing some research, I realized that the problem is with ajax requests.
Then I found the issue https://github.com/jazzband/django-debug-toolbar/pull/1610 (Automatically update History panel on AJAX requests from client.)
I don’t see any problems locally, but the error appears on the server.
It seems that the issue is that the automatic switching of History panel on AJAX does not work under certain conditions.

@tim-schilling
